app/message.py

```diff
@@ -26,6 +43,24 @@ def read(cls, message_str: str):
         return message_obj
 
 
+class MessageValidator:
+    @classmethod
+    @validate_call
+    def is_bool(cls, value: bool) -> bool:
+        return True
+
+    @classmethod
+    def validate(cls, input_type, value):
```

Comment:
what is the return type here? is `bool` but also can raise an exception?

`validate` reads in the imperative form, as such, it acts better as a command, which can fail (raising exceptions), but no need to have a return result otherwise.

The possible `bool` return, makes the `validate` method queryable.

If you want to make it queryable, I recommend to rename this to `isValid()` and not raise exceptions. My preference though, is to leave it in the imperative form and raise an exception in case is not valid. If is valid, nothing happens.

Reply:
ok I agree with your preference - the point is to raise an exception that triggers a call to the ErrorHandler. So in that case there should be no return statement?

Reply:
> So in that case there should be no return statement?

yes. No return. Just void or exception